PayPal/src/test/java/com/braintreepayments/api/PayPalLauncherUnitTest.kt

```diff
@@ -0,0 +1,124 @@
+package com.braintreepayments.api
+
+import android.content.Intent
+import android.net.Uri
+import androidx.activity.ComponentActivity
+import io.mockk.every
+import io.mockk.mockk
+import org.json.JSONException
+import org.json.JSONObject
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertSame
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class PayPalLauncherUnitTest {
+    private val browserSwitchClient: BrowserSwitchClient = mockk(relaxed = true)
+    private val activity: ComponentActivity = mockk(relaxed = true)
+    private val paymentAuthRequestParams: PayPalPaymentAuthRequestParams = mockk(relaxed = true)
+    private val intent: Intent = mockk(relaxed = true)
+    private val options: BrowserSwitchOptions = mockk(relaxed = true)
+    private val browserSwitchRequest = BrowserSwitchRequest(
+        1,
+        Uri.parse("http://"),
+        JSONObject().put("test_key", "test_value"),
+        "return-url-scheme",
+        false
+    )
```

Comment:
What are your thoughts on adding named parameters to this constructor? It might add some clarity to some of the arguments.

Reply:
Ah, i didn't realize `BrowserSwitchRequest` is a Java file.